src/main/java/org/jabref/gui/fieldeditors/LinkedEntriesEditor.java

```diff
@@ -51,21 +50,16 @@ public class LinkedEntriesEditor extends HBox implements FieldEditorFX {
 
     private final LinkedEntriesEditorViewModel viewModel;
 
-    public LinkedEntriesEditor(Field field, BibDatabaseContext databaseContext, SuggestionProvider<BibEntry> suggestionProvider, FieldCheckers fieldCheckers) {
+    public LinkedEntriesEditor(Field field, BibDatabaseContext databaseContext, SuggestionProvider<?> suggestionProvider, FieldCheckers fieldCheckers) {
         ViewLoader.view(this)
                   .root(this)
                   .load();
 
         this.viewModel = new LinkedEntriesEditorViewModel(field, suggestionProvider, databaseContext, fieldCheckers, undoManager);
 
         entryLinkField.setCellFactory(new ViewModelListCellFactory<ParsedEntryLink>().withText(ParsedEntryLink::getKey));
-        // Mind the .collect(Collectors.toList()) as the list needs to be mutable
-        entryLinkField.setSuggestionProvider(request ->
-                suggestionProvider.getPossibleSuggestions().stream()
-                                  .filter(suggestion -> suggestion.getCitationKey().orElse("").toLowerCase()
-                                                                  .contains(request.getUserText().toLowerCase()))
-                                  .map(ParsedEntryLink::new)
-                                  .collect(Collectors.toList()));
+        entryLinkField.setSuggestionProvider(request -> viewModel.getSuggestions(request.getUserText()));
+
         entryLinkField.setTagViewFactory(this::createTag);
         entryLinkField.setConverter(viewModel.getStringConverter());
         entryLinkField.setNewItemProducer(searchText -> viewModel.getStringConverter().fromString(searchText));
```

Comment:
I think there is no need for the `ItemProducer` and the `StringConverter` again because I created a new `ParsedEntryLink` with the requested text and added it to the beginning of the suggestion list in the `getSuggestions` method.